### ISSUE TITLE
yolov5-6.1版本的早停机制不起作用，设置patience=5，并不会触发早停。

### ISSUE THREAD

**Issue by sc-guihub** [ultralytics/yolov5#13708]
### Search before asking

- [x] I have searched https://github.com/ultralytics/yolov5/issues and did not find a similar report.

### Bug

yolov5-6.1版本的早停机制不起作用，设置patience=5，并不会触发早停。

### Environment

yolov5-6.1版本的早停机制不起作用，设置patience=5，并不会触发早停。

### Are you willing to submit a PR?

- [ ] Yes I'd like to help by submitting a PR!

**Comment by UltralyticsAssistant** [comment]
👋 Hello @sc-guihub, thank you for your report and for using YOLOv5 🚀! This is an automated response to help us triage your issue quickly—an Ultralytics engineer will also assist soon 🙂  
Please visit our ⭐️ [Tutorials](https://docs.ultralytics.com/yolov5/) to get started, where you can find quickstart guides for simple tasks like [Custom Data Training](https://docs.ultralytics.com/yolov5/tutorials/train_custom_data/) all the way to advanced concepts like [Hyperparameter Evolution](https://docs.ultralytics.com/yolov5/tutorials/hyperparameter_evolution/).
If this is a 🐛 Bug Report, please provide a **minimum reproducible example (MRE)** to help us debug it.
If this is a custom training ❓ Question, please provide as much information as possible, including dataset image examples and training logs, and verify you are following our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/).

To help us investigate why early stopping with `patience=5` isn’t triggering in YOLOv5 6.1, please add the following to this issue:
-Exact command used to train (full `python train.py ...` line) including where `patience` is set
-Your complete training log from start until the point you expect early stopping to trigger
-The YOLOv5 version/commit you’re using and how it was installed (clone + pip, etc.)
-Any custom changes to [training](https://github.com/ultralytics/yolov5/blob/master/train.py) (even small ones)
-A minimal dataset/config to reproduce (or a small subset + your `data.yaml`) as an MRE

## Requirements
[**Python>=3.8.0**](https://www.python.org/) with all [requirements.txt](https://github.com/ultralytics/yolov5/blob/master/requirements.txt) installed including [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/). To get started:
```bash
git clone https://github.com/ultralytics/yolov5  # clone
cd yolov5
pip install -r requirements.txt  # install
```

## Environments
YOLOv5 may be run in any of the following up-to-date verified environments (with all dependencies including [CUDA](https://developer.nvidia.com/cuda-zone)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/) and [PyTorch](https://pytorch.org/) preinstalled):
-**Notebooks** with free GPU: <a href="https://bit.ly/yolov5-paperspace-notebook"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"></a> <a href="https://colab.research.google.com/github/ultralytics/yolov5/blob/master/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolov5"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
-**Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
-**Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
-**Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/yolov5"><img src="https://img.shields.io/docker/pulls/ultralytics/yolov5?logo=docker" alt="Docker Pulls"></a>

## Status
<a href="https://github.com/ultralytics/yolov5/actions/workflows/ci-testing.yml"><img src="https://github.com/ultralytics/yolov5/actions/workflows/ci-testing.yml/badge.svg" alt="YOLOv5 CI"></a>  
If this badge is green, all [YOLOv5 GitHub Actions](https://github.com/ultralytics/yolov5/actions) Continuous Integration (CI) tests are currently passing. CI tests verify correct operation of YOLOv5 [training](https://github.com/ultralytics/yolov5/blob/master/train.py), [validation](https://github.com/ultralytics/yolov5/blob/master/val.py), [inference](https://github.com/ultralytics/yolov5/blob/master/detect.py), [export](https://github.com/ultralytics/yolov5/blob/master/export.py) and [benchmarks](https://github.com/ultralytics/yolov5/blob/master/benchmarks.py) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

**Comment by sc-guihub** [comment]
python -m torch.distributed.launch --nproc_per_node 4 train.py --weights yolov5m.pt --cfg models/yolov5m.yaml --data data/highway.yaml --epochs 300 --batch-size 64 --device 0,1,2,3 --patience 5

**Comment by pderrenger** [comment]
YOLOv5 的 `--patience` 早停是按“每个 epoch 验证集 fitness 是否有提升”来计数的，所以如果你的 `data/highway.yaml` 里没有配置 `val:`（或训练时关闭了验证），或者 fitness 仍在微小波动上升（哪怕很小），都不会触发早停；请先用最新 `master`（`git pull`）在单卡 `--device 0` 下复现一次并贴出包含每个 epoch 验证指标/fitness 的日志，以及 `highway.yaml` 里 `train/val` 路径配置，我们再一起确认是否是 v6.1 在 DDP 下的同步/停止信号问题。另可参考早停的判定逻辑说明见 [What is Overfitting? (Early Stopping)](https://www.ultralytics.com/glossary/overfitting)。
